### PR TITLE
fix(UNTICKETED): Dont parse pdf as json

### PR DESCRIPTION
**Pull request by jgordo04**
### What

PDF responses are being parsed as JSON and throwing an error. We need to stop doing that.

### CODE REVIEW

File: spec/httpigeon/request_spec.rb [hunk 1]
```diff
@@ -269,48 +272,63 @@
...
           response_body: 'invalid json',
-          expected_parsed_response: 'invalid json'
+          expected_parsed_response: 'invalid json',
+          headers: { 'content-type' => 'application/json' }
+        },
+        {
+          description: 'when the response is a pdf',
+          response_body: File.binread('spec/test-image.pdf'),
+          expected_parsed_response: File.binread('spec/test-image.pdf'),
+          headers: { 'content-type' => 'application/pdf' }
```
Comment (harry-stebbins-dailypay) [inline spec/httpigeon/request_spec.rb:324]:
```suggestion
          headers: { 'content-type' => 'application/pdf' }
        },
        {
          description: 'when no headers are set, json is assumed',
          response_body: '{ "response": "body" }',
          expected_parsed_response: { response: 'body' }.with_indifferent_access,
```

**Review by harry-stebbins-dailypay — APPROVED** [review]
consider an additional test case to "document" the behavior when the content-type is blank

**Review by eshim12 — APPROVED** [review]

File: lib/httpigeon/response.rb [hunk 2]
```diff
@@ -25,7 +25,8 @@ def [](key)
     private
 
     def parse_response
-      parsed_body = body.is_a?(String) ? JSON.parse(body) : body
+      parsable_content_type = headers['content-type'].blank? || headers['content-type'].include?('application/json')
+      parsed_body = body.is_a?(String) && parsable_content_type ? JSON.parse(body) : body
       @parsed_response = deep_with_indifferent_access(parsed_body)
```
Comment (2k-joker) [inline lib/httpigeon/response.rb:30]: This line will still fail actually

Comment (jgordo04) [reply lib/httpigeon/response.rb:30]: just pushed a fix!

Comment (2k-joker) [reply lib/httpigeon/response.rb:30]: oops nvm... I lied

**Review by therealparmesh — APPROVED** [review]

**Review by 2k-joker — APPROVED** [review]

**Pull request merged**